tests/integration/questionnaire/test_questionnaire_list_collector_primary_person.py

```diff
@@ -7,7 +7,7 @@
 
 class TestQuestionnaireListCollector(IntegrationTestCase):
     def get_link(self, row_index, text):
-        selector = f"tbody:nth-child({row_index}) td:last-child a"
+        selector = f"tbody:nth-child({row_index + 1}) td:last-child a"
```

Comment:
Using nth child wasn't ideal previously, but adding one to skip the header is even less clear.

There is a common `get_link` method in `QuestionnaireTestCase`, which should be used. To do this, you'll need to:
- change the base class of `IntegrationTestCase` to `QuestionnaireTestCase`
- change the call `self.get_link(1, "Change")` below to `self.get_link("change", 1)` 

It's also worth considering removing the `get_section_summary_link` method at the same time, as it looks like could also just use the `get_link` method.

Reply:
I replaced these methods, thanks!